src/dialects/mysql/query-generator-typescript.ts

```diff
@@ -14,6 +16,63 @@ export class MySqlQueryGeneratorTypeScript extends AbstractQueryGenerator {
     return `SHOW FULL COLUMNS FROM ${this.quoteTable(tableName)};`;
   }
 
+  showConstraintsQuery(tableName: TableNameOrModel, constraintName?: string) {
+    const table = this.extractTableDetails(tableName);
+    const dbVersion = this.sequelize.getDatabaseVersion() || '0.0.0';
+
+    // MySQL 8.0.16+ has a new INFORMATION_SCHEMA.CHECK_CONSTRAINTS table
```

Comment:
Looking at the integration tests it seems that we have a different behaviour depending on the used databaseVersion. Would it be possible to have the same behaviour on MySQL 5.7 and 8.0.0-8.0.15? I'd prefer not to have different queries depending on the databaseVersion

Reply:
The issue is that in MySQL version 8.0.16+ has a new INFORMATION_SCHEMA.CHECK_CONSTRAINTS table which allows the `showConstraintsQuery` to list the check constraints however in versions prior, this is not possible and the query will return an error.